### ISSUE TITLE
Cannot add a previously removed external reference by DOI, without a page reload 🐛 

### ISSUE THREAD

**Issue by nsunami** [libscie/ResearchEquals.com#747]
### URL of the Page

https://www.researchequals.com/drafts

### What happened?

When drafting a module, I could not add a reference by DOI that I removed previously without a reload.

After a full reload of the page, entering a DOI will show the added research output.

https://user-images.githubusercontent.com/17035406/192352948-0822f651-f01c-4374-8dc8-ba9519675edb.mov

### Steps to reproduce

1. Create a new draft
2. Add a new research output via DOI (one that's not in R='s database) as a linked module
3. Select "Links to..."
4. Delete the added research output
5. Enter the DOI of the research output again

### What browser(s) did you use?

Chrome

### Anything else?

This is a tiny bug, but it can be confusing if the user repeatedly adds and removes the modules. 

Noticed this bug while working on #625

**Comment by nsunami** [comment]
While working on this, I found that the `createReferenceModule` mutation will try using CrossRef first, and if that fails, try to use DataCite. If there's an error in mutation (for example, violation of the unique constraint of suffix, that would lead to a hard-to-debug error, since the resulting error message will be from DataCite.

**Comment by nsunami** [comment]
After doing some trial and error, I started to think addressing this issue may not be a priority, given the small amount of improvement over a considerable amount of effort. 

I also found that the following warnings and issues makes it hard to debug this issue:

1. Warning about the React 18 issue 
2. Warning about importing `createRoot` from `react-dom`, instead of from `react-dom/client`
3. The Algolia Autocomplete window closing, due to re-rendering. (Previously mentioned in #326)

Probably, addressing any of these issues before moving forward may make sense.
